### PR TITLE
properly handle changing ownership of dependant case

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?179148
@czue 

cc @NoahCarnahan

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py [hunk 1]
```diff
@@ -29,3 +29,24 @@ def test_update_dependent_case(self):
...
+                CaseState(case_id="c1", indices=[
+                    CommCareCaseIndex(identifier="d1-id", referenced_id="d1")
+                ])
+            ],
+            dependent_cases_on_phone=[dependent_case_state],
+            owner_ids_on_phone=['user1']
+        )
+        xform_id = uuid.uuid4().hex
+        xform = XFormInstance(_id=xform_id)
+        form_actions = [CommCareCaseAction(action_type=CASE_ACTION_UPDATE, updated_known_properties={'owner_id': 'user2'})]
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py]: E501 line too long (123 > 115 characters)

**Comment by NoahCarnahan** [comment]
Looks good to me, but sounds like @czue has more context, so I'll let him merge.

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py [hunk 2]
```diff
@@ -29,3 +29,24 @@ def test_update_dependent_case(self):
             # this test just ensures it doesn't still do that.
             for log in [sync_log, SimplifiedSyncLog.from_other_format(sync_log)]:
                 log.update_phone_lists(xform, [parent_case])
+
+    def test_update_dependent_case_owner_still_present(self):
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py]: looks like this test only runs on the old code, so curious why you bothered with the `test_only_get_dependent_cases_on_phone` thing. also - have you verified this works ok on new code?

**Comment by czue** [comment]
my comment isn't blocking so @NoahCarnahan or @snopoke feel free to merge if i'm not online by the time there's a response

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py [hunk 3]
```diff
@@ -29,3 +29,24 @@ def test_update_dependent_case(self):
...
+            dependent_cases_on_phone=[dependent_case_state],
+            owner_ids_on_phone=['user1']
+        )
+        xform_id = uuid.uuid4().hex
+        xform = XFormInstance(_id=xform_id)
+        form_actions = [CommCareCaseAction(action_type=CASE_ACTION_UPDATE, updated_known_properties={'owner_id': 'user2'})]
+        with patch.object(CommCareCase, 'get_actions_for_form', return_value=form_actions):
+            parent_case = CommCareCase(_id='d1')
+            # before this test was added, the following call raised a ValueError on legacy logs.
+            for log in [sync_log, SimplifiedSyncLog.from_other_format(sync_log)]:
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py]: @czue doesn't this make it run on both old and new synclogs?

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_log_assertions.py]: sorry i'm an idiot :(

**Pull request merged**